docs/creating-a-runtime-check.md

```diff
@@ -0,0 +1,209 @@
+
+Both these interfaces are implemented in the `Abstract_Runtime_Check` class, which developers can use when building out their own runtime checks. This class defines both the `prepare()` and `run()` methods which are required to be implemented for every runtime check.
+
+Below is the basic scaffold when creating a custom runtime check.
+
+```php
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Checks\Abstract_Runtime_Check;
+use WordPress\Plugin_Check\Traits\Stable_Check;
+
+class Custom_Check extends Abstract_Runtime_Check {
+
+  use Stable_Check;
+  
+  public function get_categories() {
+    // Return an array of check categories.
+  }
+
+  public function prepare() {
+    // Handle the Checks preparations and return a cleanup function.
+  }
+
+  public function run( Check_Result $result ) {
+    // Handle running the check and adding warnings or errors to the result.
+  }
+}
+```
+
+## Preparations
+
+Preparations in the Plugin Checker are used to set up the environment before running checks, involving logic like activating themes or plugins and creating necessary test content. They utilize the `Preparation` interface, requiring the implementation of a `prepare()` method, and return cleanup functions to revert changes made during preparation. Three different approaches can be used for creating preparations based on the circumstances.
+
+### Check Preperations
+
+Check preparations are created using the `prepare()` method within a check class.
+
+This type of preparation is useful for carrying out the logic to prepare the environment that is specific to the check that implements it.
+
+Below is an example of a preparation that creates a test post and returns a cleanup function that deletes that post after the check is run.
+
+```php
+use WordPress\Plugin_Check\Checker\Checks\Abstract_Runtime_Check;
+
+class Custom_Check extends Abstract_Runtime_Check {
+  /**
+   * Check Preparation.
+   */
+  public function prepare() {
+    // Create a test post ahead of running the check.
+    $post_id = wp_insert_post(
+      array(
+        'post_title'   => 'Test post',
+        'post_content' => 'This is a test post.',
+        'post_type'    => 'post',
+      )
+    );
+
+    // Return the clean up function.
+    return function() use ( $post_id ) {
+      // Remove the test post created.
+      wp_delete_post( $post_id );
+    }
+  }
+}
+```
+
+### Preparation Classes
+
+Preparation classes are individual classes created to handle specific preparation logic in a single place.
+
+The [`Force_Single_Plugin_Preparation`](https://github.com/10up/plugin-check/blob/trunk/includes/Checker/Preparations/Force_Single_Plugin_Preparation.php) in the Plugin Checker is an example of a preparation class.
+
+Preparation classes are useful for implementing the same preparation logic across multiple checks using the concept of shared preparations.
+
+### Shared Preparations
```

Comment:
See above, these two sections should be combined into a single section "Shared preparations". Essentially, you have to write a preparation class, and then any check can (re-)use it, so that's why it is called a shared preparation. And of course there are already preparation classes in the Plugin Checker itself, which can be used as a shared preparation.

Also see https://github.com/10up/plugin-check/pull/148#discussion_r1204839818 where I mentioned this earlier.

Reply:
Thanks @felixarntz I've combined these section and added additional detail